reference/interbit-ui-components/CONTRIBUTING.adoc

```diff
@@ -0,0 +1,187 @@
+| SCSS files from React Bootstrap. These files should not be edited.
+
+| `src/scss/interbit/`
+| SCSS files for Interbit styles. Each component has its own `scss` file which
+is imported in `interbit.scss`. Style overrides for React Bootstrap components
+are also located in this folder.
+
+| `src/tests/exports.tests.js`
+| A set of simple tests to check that exported components exist.
+
+| `src/tests/uiKit.test.js`
+| Rendering tests for UI Kit components.
+
+| `src/styleguide/`
+| Build files for React Styleguidist.
+
+| `index.js`
+| Export statements for components.
+
+| `styleguide.config.js`
+| Configuration file for React Styleguidist.
+
+|===
+
+
+== Creating components
+
+=== Code structure
+
+We are in the process of defining the code structure for
+Interbit UI components. At this time, we ask that contributing developers use
+the existing code as an example of how to structure your component's code.
+
+Simple, presentational components should go in the `src/components/UIKit`
+directory. More complex components should be in their own directory within
+`src/components`. Each component must be exported from `src/index.js`.
+
+=== SCSS
+
+For each component:
+
+- The styles should be in their own file, say `_my-component.scss`, in the
+`src/scss/interbit` directory. This file should be imported in
+`src/scss/interbit.scss`.
+- The styles should be contained in a class named after the component and
+prefixed with `ibweb-`. For example, `MyComponent` should have a top-level
+class called `ibweb-my-component`. See `src/scss/interbit/_footer.scss` for how
+styles should be nested.
+- The CSS must be supported by the latest versions of Chrome, Firefox, Safari,
+and Edge. If in doubt, see link:https://caniuse.com/[caniuse.com].
+
+=== Image assets
+
+Image assets such as icons and illustrations should be placed in `src/assets` and should satisfy the following:
+
+- Icons files are placed in `src/assets/icons`.
+- Image asset files are in `svg` format wherever possible.
+- Files should not exceed 25KB in size at this time.
+
+=== Tests
+
+Components must have an export test in `src/tests/exports.test.js` for each
+(sub)-component exported from `src/index.js`. This verifies that the component
+is in fact exported, and is usable externally.
+
+UI Kit components must have, at the minimum, rendering tests in
+`src/tests/uiKit.test.js`. Interbit UI components uses Jest and Enzyme for
+testing and rendering components.
+
+=== Documentation
+
+Interbit pulls documentation out of JSDoc style code comments. For each
+component:
+
+- Give a brief description of the component before the class declaration.
+- Give a description of each prop that the component accepts
+
+For example, the `LinkBar` component has the following code comments.
+
+[source,js]
+----
+{% include "linkBar.js" %}
+----
+
+Each UI Kit component should be accompanied by an example file,
+`MyComponent.md`. link:https://react-styleguidist.js.org/docs/documenting.html[React Styleguidist]
```

Comment:
The `React Styleguidist` sentence is awkward. While the link label identifies the package, the link targets a specific aspect of the package. I'd suggest changing the link to the main `react-styleguidist` site, and then add another sentence:

```
See React Styleguidist's link:https://react-styleguidist.js.org/docs/documenting.html[Documenting]
guidelines for details.
```